Fix false-positive on timeouts attribute in aws_route_table and aws_default_route_table

chdorner-snyk opened the pull request:
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   |no
| ⚠ Deprecations?   |no
| ❌ BC Break       |no
| 🔗 Related issues | CFG-1974
| ❓ Documentation  | no

## Description

Remove `timeout` attributes for `aws_route_table` and `aws_default_route_table` resources to prevent false-positive drift.
Note that the `timeout` block for these two Terraform resources have only been added in AWS provider version 3.62.0

chdorner-snyk committed:
chore: Allow to set provider version in tests

Different AWS terraform state reader resource tests can use different provider versions.

chdorner-snyk committed:
chore: Use provider 3.62 for AWS route table tests

This is the version that introduces support for the `timeout` block which we'll filter out in a next commit.

chdorner-snyk committed:
chore: Add AWS provider 3.62 test schema

chdorner-snyk committed:
fix: Filter out timeouts from aws_route_table

Configurable timeouts on the aws_route_table resource are Terraform-only, thus filtering them out prevents false-positive drift to be shown.

chdorner-snyk committed:
test: Use timeouts in AWS routetable acc test

chdorner-snyk committed:
fix: Remove timeouts from aws_default_route_table

eliecharra commented on pkg/iac/terraform/state/test/aws_default_route_table/terraform.tfstate:
```diff
@@ -0,0 +1,108 @@
+    },
+    {
+      "mode": "managed",
+      "type": "aws_default_vpc",
```
You can remove everything but aws_route_table resource from that state file. It will be an inconsistent one regarding Terraform logic but we do not care here it's only to ensure unmarshalling is going well !
We try to minimise test maintenance by removing side effects 😉 

You can also remove the lock file from that folder, it is not used.

You'll also have to update the result file.

eliecharra replied:
> You can also remove the lock file from that folder, it is not used.

☝🏻 `pkg/iac/terraform/state/test/aws_default_route_table/.terraform.lock.hcl` need to be removed too

eliecharra commented on pkg/resource/aws/aws_route_table_test.go:
```diff
@@ -9,7 +9,7 @@ import (
 func TestAcc_Aws_RouteTable(t *testing.T) {
 	acceptance.Run(t, acceptance.AccTestCase{
-		TerraformVersion: "0.15.5",
+		TerraformVersion: "1.0.2",
```
I think it's perfectly fine if you update test as you reach them for M1 support 👍🏻 
We do not want a specific version of Terraform here, it's just to pin a version and prevent flackyness in tests due to Terraform changes.

I'm totally happy with that no worries 🙏🏻

eliecharra reviewed (changes requested):
Thanks a lot for that PR that looks awesome 🎉 

I only have small remark on IaC test 

Super good job !

chdorner-snyk committed:
chore: Remove additional resources from fixtures

There's no need to keep i.e. the VPCs in the fixtures for parsing the route table TF state files.

eliecharra reviewed (approved):
Sounds great 🎉  ! Thanks a lot @chdorner-snyk 🙏🏻
